### PR TITLE
Upgrade to Laravel 10 [API-382]

### PR DESCRIPTION
**Pull request by nikhiltri**

### CODE REVIEW

File: app/Console/Kernel.php [hunk 1]
```diff
@@ -23,11 +23,12 @@ class Kernel extends ConsoleKernel
 
     /**
      * Define the application's command schedule.
-     *
-     * @return void
      */
-    protected function schedule(Schedule $schedule)
+    protected function schedule(Schedule $schedule): void
     {
+        $schedule->command('cache:prune-stale-tags')
```
Comment (zachgarwood) [inline app/Console/Kernel.php:29]: Why the need to change which scheduled command is run?

File: app/Providers/RouteServiceProvider.php [hunk 2]
```diff
@@ -11,22 +11,22 @@
...
-     *
-     * @return void
      */
-    public function boot()
+    public function boot(): void
     {
-        $this->configureRateLimiting();
+        RateLimiter::for('api', function (Request $request) {
+            return Limit::perMinute(60)->by($request->user()?->id ?: $request->ip());
+        });
```
Comment (zachgarwood) [inline app/Providers/RouteServiceProvider.php:29]: 👍

**Review by zachgarwood — COMMENTED** [review]
I left a question about one of the changes.

File: app/Console/Kernel.php (hunk 1 above)
Comment (nikhiltri) [reply app/Console/Kernel.php:29]: Not a change here, but an addition of the `cache:prune-stale-tags` command based on new Redis functionality. More here: https://laravel.com/docs/10.x/upgrade#redis-cache-tags

**Pull request merged**